### PR TITLE
Text: Add breakAll option to Text

### PR DESCRIPTION
**Pull request by rlingineni**
### Summary

Text didn't support a `BreakAll` option. 

And lineClamp by default only supported `breakWord` It was in the added in [this PR](https://github.com/pinterest/gestalt/pull/1565), and then later changed to became a [break-word](https://github.com/pinterest/gestalt/pull/1577).

We do have a few instances in Pinboard where LineClamp is used with `overflow="normal"`, but the word-break is actually `break-word` because it's used alongside lineClamp. This PR makes the `breakWord` its own class, and it is added by default, alongside lineClamp to preserve existing behavior.

However, it also adds support for `break-all` as an option to be used alongside `lineClamp`

#### Why?

Request here in [Slack](https://pinterest.slack.com/archives/C13KLG5P0/p1690853063014349). To support `lineClamp` with breakAll

Before:
image

After:
image

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-XXXX)
- [TDD](link to Paper doc)
- [Figma](link to Figma file)

### Checklist

- [ ] Added unit and Flow Tests
- [ ] Added documentation + accessibility tests
- [ ] Verified accessibility: keyboard & screen reader interaction
- [ ] Checked dark mode, responsiveness, and right-to-left support
- [ ] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

File: packages/gestalt/src/Typography.css [hunk 1]
```diff
@@ -75,7 +75,11 @@
 /* overflow */
 
 .breakWord {
-  word-wrap: break-word;
```
Comment (AlbertCarreras) [inline packages/gestalt/src/Typography.css:78]: word-wrap: break-word; & word-break: break-word; are equivalent. can we add that to the description?

https://www.w3schools.com/cssref/playdemo.php?filename=playcss_word-break&preval=break-word
https://www.w3schools.com/cssref/playdemo.php?filename=playcss_word-wrap&preval=break-word

File: packages/gestalt/src/Text.js [hunk 2]
```diff
@@ -111,12 +117,13 @@ const TextWithForwardRef: AbstractComponent<Props, HTMLElement> = forwardRef<Pro
...
       align === 'forceRight' && typography.alignForceRight,
       overflow === 'breakWord' && typography.breakWord,
+      overflow === 'breakAll' && typography.breakAll,
       overflow === 'noWrap' && typography.noWrap,
       italic && typography.fontStyleItalic,
       underline && typography.underline,
       weight === 'bold' && typography.fontWeightSemiBold,
       weight === 'normal' && typography.fontWeightNormal,
-      isNotNullish(lineClamp) && typography.lineClamp,
+      lineClampStyles,
```
Comment (AlbertCarreras) [inline packages/gestalt/src/Text.js:126]: could we refactor to prevent overrides?

if overflow === 'breakWord' and isLineClamp technically we would be setting breakWord twice
overflow === 'breakWord' && typography.breakWord
&
[typography.breakWord]: isNotNullish(lineClamp) && overflow !== 'breakAll',

Maybe?
`
const isLineClamp = isNotNullish(lineClamp)
...
overflow === 'breakAll' && typography.breakAll,      
{
[typography.breakWord]: overflow === 'breakWord' ||  (isLineClamp && overflow !== 'breakAll')
[typography.lineClamp]: isLineClamp
}
`

or 

...
overflow === 'breakAll' && typography.breakAll, 
(overflow === 'breakWord' ||  (isLineClamp && overflow !== 'breakAll')) && typography.breakWord,
isLineClamp && typography.lineClamp
`

File: packages/gestalt/src/Text.js [hunk 3]
```diff
@@ -11,7 +11,7 @@ function isNotNullish(val: ?number): boolean {
 }
 
 type As = 'span' | 'div';
-type Overflow = 'normal' | 'breakWord' | 'noWrap';
+type Overflow = 'normal' | 'breakAll' | 'breakWord' | 'noWrap';
```
[the remark this answers is not in the document]
Comment (AlbertCarreras) [reply packages/gestalt/src/Text.js:14]: Mostly bacause of the issue shown 

Line clamp truncationExample in CodeSandbox; Opens a new tabWhen the "lineClamp" prop is set to 1, text with long strings of characters might be truncated incorrectly.
--


[Example in CodeSandbox; Opens a new tab](https://codesandbox.io/s/development-mode-forked-44qkwk?file=/example.js)
When the "lineClamp" prop is set to 1, text with long strings of characters might be truncated incorrectly.


Can we chek if we are fixing this issue? and maybe removing the known issue ?

**Review by AlbertCarreras — APPROVED** [review]
LGTM but left several comments, but overall looks good

File: packages/gestalt/src/Text.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (rlingineni) [reply packages/gestalt/src/Text.js:14]: Yes, we should

File: packages/gestalt/src/Typography.css (hunk 1 above)
Comment (rlingineni) [reply packages/gestalt/src/Typography.css:78]: I don't think we should use word wrap, it's actually a deprecated property:
https://stackoverflow.com/questions/17259916/difference-between-overflow-wrap-and-word-break

File: packages/gestalt/src/Text.js (hunk 2 above)
Comment (rlingineni) [reply packages/gestalt/src/Text.js:126]: I agree, the overrides can be hard to follow. I was thinking through this, but I thought it was super hard to read/think through.

I refactored to a method like this:
```js
 const getOverflowStyle = (): string | void => {
      if (overflow === 'noWrap') {
        return typography.noWrap;
      }

      if (overflow === 'breakAll') {
        return typography.breakAll;
      }

      // default to breakWord if lineClamp is set
      if (overflow === 'breakWord' || isNotNullish(lineClamp)) {
        return typography.breakWord;
      }

      return undefined;
    };
```
